engine/src/flutter/lib/web_ui/lib/src/engine/safe_browser_api.dart

```diff
@@ -56,6 +56,40 @@ num? parseFontSize(DomElement element) {
   return fontSize;
 }
 
+/// Parses the given style property [attributeName] of [element] and returns the
+/// value without a unit.
+num? parseStyleProperty(DomElement element, String attributeName) {
+  num? styleProperty;
+
+  if (element.has('computedStyleMap')) {
+    final JSAny? rawStyleProperty = element
+        .computedStyleMap()
+        .get(attributeName)
+        ?.getProperty<JSAny>('value'.toJS);
+    // If the captured rawStyleProperty is not a JSNumber we should fallback
+    // to `getComputedStyle`. This covers cases when the style property
+    // maybe be set to keywords like 'normal', the `computedStyleMap` does
+    // not capture these as absolute values.
+    if (rawStyleProperty != null && rawStyleProperty.isA<JSNumber>()) {
+      styleProperty = (rawStyleProperty as JSNumber).toDartDouble;
+    }
+  }
+
+  // Fallback to `getComputedStyle` if the first attempt fails.
+  if (styleProperty == null) {
+    final String stylePropertyString = domWindow
+        .getComputedStyle(element)
+        .getPropertyValue(attributeName);
+
+    final num? parsed = parseFloat(stylePropertyString);
+    if (parsed != null && !parsed.isNaN) {
+      styleProperty = parsed;
+    }
```

Comment:
```suggestion
    styleProperty = parsed;
```
The extra checks are not necessary. `parseFloat()` already handles the `isNaN` case.